src/proof.rs

```diff
@@ -494,6 +490,7 @@ impl<'a, 'b> ProofTreePrinterImpl<'a, 'b> {
 
     fn init_stmt_lookup(&mut self) {
         for tree in &self.p.arr.trees {
+            #[allow(clippy::map_entry)]
```

Comment:
I haven't tested this, but the usual way to fix this is to match on `self.stmt_lookup.entry(tree.address)` and call `e.insert()` in the `Vacant(e)` case. Just looking at this code though, I don't see anything else touching `self.stmt_lookup` between the `contains_key` and the `insert`, so where is the double borrow coming from?